mmdet3d/models/fusion_layers/vote_fusion.py

```diff
@@ -0,0 +1,251 @@
+import torch
+from torch import nn as nn
+
+from mmdet3d.core.bbox import Coord3DMode, points_cam2img
+from ..registry import FUSION_LAYERS
+from . import Coord3DTransformation
+
+
+@FUSION_LAYERS.register_module()
+class VoteFusion(nn.Module):
+    """Fuse 2d features from 3d seeds.
+
+    Args:
+        num_classes (int): number of classes.
+        img_norm_cfg (dict): image normalization config.
+        max_imvote_per_pixel (int): max number of imvotes.
+    """
+
+    def __init__(self,
+                 num_classes=10,
+                 img_norm_cfg=None,
+                 max_imvote_per_pixel=3):
+        super(VoteFusion, self).__init__()
+        self.num_classes = num_classes
+        self.img_norm_cfg = img_norm_cfg
+        self.max_imvote_per_pixel = max_imvote_per_pixel
+        assert isinstance(self.img_norm_cfg, dict)
+        assert 'mean' in self.img_norm_cfg.keys()
+
+    def forward(self, imgs, bboxes_2d_rescaled, seeds_3d_depth, img_metas,
+                calibs):
+        """Forward function.
+
+        Args:
+            imgs (list[torch.Tensor]): Image features.
+            bboxes_2d_rescaled (list[torch.Tensor]): 2D bboxes.
+            seeds_3d_depth (torch.Tensor): 3D seeds.
+            img_metas (list[dict]): Meta information of images.
+            calibs: Camera calibration information of the images.
+
+        Returns:
+            torch.Tensor: Concatenated cues of each point.
+            torch.Tensor: Validity mask of each feature.
+        """
+        img_features = []
+        masks = []
+        for i, data in enumerate(
+                zip(imgs, bboxes_2d_rescaled, seeds_3d_depth, img_metas)):
+            img, bbox_2d_rescaled, seed_3d_depth, img_meta = data
+            bbox_num = bbox_2d_rescaled.shape[0]
+            seed_num = seed_3d_depth.shape[0]
+
+            # xyz_depth = seed_3d_depth.clone()
+            bbox_2d_origin = bbox_2d_rescaled.clone().float()
+            img_shape = img_meta['img_shape']
+            ori_shape = img_meta['ori_shape']
+            # origin bboxes
+            img_h, img_w, _ = img_shape
+            ori_h, ori_w, _ = ori_shape
+
+            # set up coords transformation
+            coords_trans = Coord3DTransformation(seed_3d_depth.dtype,
+                                                 seed_3d_depth.device, 'DEPTH',
+                                                 img_meta)
+
+            img_scale_factor = (
+                seed_3d_depth.new_tensor(img_meta['scale_factor'][:2])
+                if 'scale_factor' in img_meta.keys() else [1.0, 1.0])
+
+            img_flip = img_meta['flip'] if 'flip' in \
+                img_meta.keys() else False
+            img_crop_offset = (
+                seed_3d_depth.new_tensor(img_meta['img_crop_offset'])
+                if 'img_crop_offset' in img_meta.keys() else 0)
+
+            # first reverse the data transformations
+            xyz_depth = coords_trans(seeds_3d_depth, 'HTSR', reverse=True)
```

Comment:
This is a little bit hard code. If necessary, the img_meta should contain the information of 'HTSR'